File: chall01/mbenjell.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   mbenjell.c                                         :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: mbenjell <mbenjell@student.42.fr>          +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2020/04/13 14:02:40 by mbenjell          #+#    #+#             */
/*   Updated: 2020/04/13 14:52:02 by mbenjell         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <stdlib.h>

char					*ft_rgb2hex(int r, int g, int b)
{
	static const char	l[16] = "0123456789abcdef";
	char				*buff = malloc(8);

	buff[0] = '#';
	buff[8] = 0;
	int i = 6;
	int hex = (r << 16) + (g << 8) + b;
	while (i > 0)
	{
		buff[i--] = l[hex % 16];
		hex /= 16;
	}
	return (buff);
}

File: chall02/mbenjell.py
#!/usr/bin/env python3

# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    mbenjell.py                                        :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: mbenjell <mbenjell@student.42.fr>          +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2020/04/21 02:28:42 by mbenjell          #+#    #+#              #
#    Updated: 2020/04/21 04:21:53 by mbenjell         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

import sys
import re

if __name__ == '__main__':
    
    regex = r"^[a-zA-Z ]+$"

    if len(sys.argv) != 2 or re.match(regex, sys.argv[1]) is None:
        print('usage: ./mbenjell.py <a-zA-Z string>')
        exit()

    string = sys.argv[1]
    morse = ""
    data = ".- -... -.-. -.. . ..-. --. .... .. .--- -.- .-.. -- -. --- .--. --.- .-. ... - ..- ...- .-- -..- -.-- --.." 

    string = string.lower()
    list = data.split(' ')
    list.append(' ')
    alphabet = "abcdefghijklmnopqrstuvwxyz "
    for l in string:
        morse += list[alphabet.find(l)] 
    print(morse)
